File: sdk/dart/lib/model/approval_rule.dart
part of swagger.api;


@Entity()
class ApprovalRule {
  
  @Property(name: 'ID')
  String ID = null;
  

  @Property(name: 'Name')
  String name = null;
  

  @Property(name: 'Description')
  String description = null;
  

  @Property(name: 'ApprovingGroupID')
  String approvingGroupID = null;
  

  @Property(name: 'RuleExpression')
  String ruleExpression = null;
  

  @Property(name: 'Scope')
  String scope = null;
  

  @Property(name: 'ScopeTimeUnit')
  String scopeTimeUnit = null;
  

  @Property(name: 'ScopeTimeNumber')
  int scopeTimeNumber = null;
  

  @Property(name: 'ScopeStartDate')
  DateTime scopeStartDate = null;
  

  @Property(name: 'ExpireAfterTimeUnit')
  String expireAfterTimeUnit = null;
  

  @Property(name: 'ExpireAfterNumber')
  int expireAfterNumber = null;
  

  @Property(name: 'ApproveOnExpire')
  bool approveOnExpire = null;
  

  @Property(name: 'xp')
  Object xp = null;
  
  ApprovalRule();

  @override
  String toString()  {
    return 'ApprovalRule[ID=$ID, name=$name, description=$description, approvingGroupID=$approvingGroupID, ruleExpression=$ruleExpression, scope=$scope, scopeTimeUnit=$scopeTimeUnit, scopeTimeNumber=$scopeTimeNumber, scopeStartDate=$scopeStartDate, expireAfterTimeUnit=$expireAfterTimeUnit, expireAfterNumber=$expireAfterNumber, approveOnExpire=$approveOnExpire, xp=$xp, ]';
  }

}

File: sdk/javascript-closure-angular/API/Client/ApprovalRule.js
goog.provide('API.Client.ApprovalRule');

/**
 * @record
 */
API.Client.ApprovalRule = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.ApprovalRule.prototype.ID;

/**
 * @type {!string}
 * @export
 */
API.Client.ApprovalRule.prototype.name;

/**
 * @type {!string}
 * @export
 */
API.Client.ApprovalRule.prototype.description;

/**
 * @type {!string}
 * @export
 */
API.Client.ApprovalRule.prototype.approvingGroupID;

/**
 * @type {!string}
 * @export
 */
API.Client.ApprovalRule.prototype.ruleExpression;

/**
 * @type {!string}
 * @export
 */
API.Client.ApprovalRule.prototype.scope;

/**
 * @type {!string}
 * @export
 */
API.Client.ApprovalRule.prototype.scopeTimeUnit;

/**
 * @type {!number}
 * @export
 */
API.Client.ApprovalRule.prototype.scopeTimeNumber;

/**
 * @type {!API.Client.date}
 * @export
 */
API.Client.ApprovalRule.prototype.scopeStartDate;

/**
 * @type {!string}
 * @export
 */
API.Client.ApprovalRule.prototype.expireAfterTimeUnit;

/**
 * @type {!number}
 * @export
 */
API.Client.ApprovalRule.prototype.expireAfterNumber;

/**
 * @type {!boolean}
 * @export
 */
API.Client.ApprovalRule.prototype.approveOnExpire;

/**
 * @type {!API.Client.Object}
 * @export
 */
API.Client.ApprovalRule.prototype.xp;

